src/v/security/config.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+
+#include "base/seastarx.h"
+
+#include <seastar/core/sstring.hh>
+
+#include <optional>
+#include <vector>
+
+namespace security::tls {
+
+std::optional<ss::sstring>
+validate_rules(const std::optional<std::vector<ss::sstring>>& r) noexcept;
+
+}
+
+namespace security::oidc {
+
+std::optional<ss::sstring>
+validate_principal_mapping_rule(ss::sstring const& rule);
```

Comment:
Where are these defined?

Reply:
Looks like ./security/config_bsl.cc. Should this be included there?

Reply:
it's in config_rsl.cc, but i'm not sure what you mean by included "there"--this is header file declaration.

Reply:
Ah sorry I wasn't clear. I would've thought config_rsl.cc should include this new header. I guess it's a little special since it's in its own security_config library.

Reply:
If the compiler is happy, then I'm happy! But I just added a commit to add those headers because its a little weird to not have it as you point out.